components/script/dom/xmlhttprequest.rs

```diff
@@ -979,14 +979,12 @@ impl XMLHttpRequest {
                 // Part of step 11, send() (processing response end of file)
                 // XXXManishearth handle errors, if any (substep 2)
 
-                // Subsubsteps 5-7
+                // Subsubsteps 6-8
                 self.send_flag.set(false);
 
                 self.change_ready_state(XMLHttpRequestState::Done);
                 return_if_fetch_was_terminated!();
-                // Subsubsteps 10-12
-                self.dispatch_response_progress_event(atom!("progress"));
```

Comment:
I'm really sorry @KiChjang but I don't understand why we need to check the ContentLength?  As I've explained above, we are sure that the ```progress``` event is dispatched by [this line](https://github.com/servo/servo/blob/master/components/script/dom/xmlhttprequest.rs#L970). So when we are asynchronous, we don't need to fire an event at the *Done* step because each time we add data, a progress event is already fired. 

On the other hand, i think that checking if we are synchronous to fire a ```progress``` event could be nice. Like this:
```rust
if (self.sync.get()) {
    self.dispatch_response_progress_event(atom!("progress"));
}
```